### PR TITLE
clean up testing to not include gtest's tests

### PR DESCRIPTION
**Pull request by ouj**
Use `EXCLUDE_FROM_ALL` to not run tests on `all` target

### CODE REVIEW

**Comment by ouj** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10** <a href="https://app.graphite.dev/github/pr/loomhq/muxers/10" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/loomhq/muxers/10?utm_source=stack-comment).

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,11 +36,14 @@ FetchContent_Declare(
   URL https://github.com/google/googletest/archive/refs/tags/release-1.12.0.zip
   DOWNLOAD_EXTRACT_TIMESTAMP YES
 )
-# For Windows: Prevent overriding the parent project's compiler/linker settings
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-if(APPLE)
-  set(CMAKE_GTEST_DISCOVER_TESTS_DISCOVERY_MODE PRE_TEST)
+if(NOT googletest_POPULATED)
```
Comment (vhmth) [inline CMakeLists.txt:39]: What does googletest do for us? We're just using it as a testing framework right? Also, what does `googletest_POPULATED` actually mean? Tried googling (lol) for it

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,9 +17,7 @@ project(
   LANGUAGES CXX C
 )
 
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```
Comment (vhmth) [inline CMakeLists.txt:20]: Why did we have this check before?

File: CMakeLists.txt (hunk 1 above)
Comment (ouj) [reply CMakeLists.txt:39]: The unit tests are written using great.

Populated meant it is downloaded added as subdirectory.

File: CMakeLists.txt (hunk 2 above)
Comment (ouj) [reply CMakeLists.txt:20]: Because of some random advice from the internet when I did know what they meant. 😂

**Review by gchilds — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (vhmth) [reply CMakeLists.txt:20]: 😅👌🏾

**Pull request merged**